pkg/network/kubemacpool_test.go

```diff
@@ -0,0 +1,184 @@
+				Expect(errorList).To(BeEmpty())
+			})
+		})
+
+	})
+
+	var _ = Describe("Testing fillDefaultsKubeMacPool", func() {
+		Context("When kubeMacPool is nil", func() {
+			It("should NOT return an error and do nothing", func() {
+				currentClusterConfig := &opv1alpha1.NetworkAddonsConfigSpec{}
+				previousClusterConfig := &opv1alpha1.NetworkAddonsConfigSpec{
+					KubeMacPool: &opv1alpha1.KubeMacPool{RangeStart: "02:00:00:00:00:00", RangeEnd: "0A:FF:FF:FF:FF:FF"}}
+				errorList := fillDefaultsKubeMacPool(currentClusterConfig, previousClusterConfig)
+				Expect(errorList).To(BeEmpty())
+			})
+		})
+		Context("When the user hasn't explicitly requested a range", func() {
+			Context("When a previous kubeMacPool exits", func() {
+				It("should use the previous kubeMacPoll range for the current one, and not return an error", func() {
+					previousClusterConfig := &opv1alpha1.NetworkAddonsConfigSpec{
+						KubeMacPool: &opv1alpha1.KubeMacPool{RangeStart: "02:00:00:00:00:00", RangeEnd: "0A:FF:FF:FF:FF:FF"}}
+					currentClusterConfig := &opv1alpha1.NetworkAddonsConfigSpec{
+						KubeMacPool: &opv1alpha1.KubeMacPool{RangeStart: "", RangeEnd: ""}}
+					errorList := fillDefaultsKubeMacPool(currentClusterConfig, previousClusterConfig)
+					Expect(errorList).To(BeEmpty())
+					Expect(currentClusterConfig.KubeMacPool.RangeStart).To(Equal(previousClusterConfig.KubeMacPool.RangeStart))
+					Expect(currentClusterConfig.KubeMacPool.RangeEnd).To(Equal(previousClusterConfig.KubeMacPool.RangeEnd))
+
+				})
+			})
+			Context("When a previous kubeMacpool doesn't exits", func() {
+				It("should generate a new range for the current kubeMacPoll, and not return an error", func() {
+					previousClusterConfig := &opv1alpha1.NetworkAddonsConfigSpec{}
+					currentClusterConfig := &opv1alpha1.NetworkAddonsConfigSpec{
+						KubeMacPool: &opv1alpha1.KubeMacPool{RangeStart: "", RangeEnd: ""}}
+					errorList := fillDefaultsKubeMacPool(currentClusterConfig, previousClusterConfig)
+					Expect(errorList).To(BeEmpty())
+					Expect(currentClusterConfig.KubeMacPool.RangeStart).To(Not(Equal("")), "RangeStart:")
+					Expect(currentClusterConfig.KubeMacPool.RangeEnd).To(Not(Equal("")), "RangeEnd:")
+
+				})
+			})
+
+		})
+
+		Context("When the user has explicitly requested a range", func() {
+			It("should leave the range as it is, and not return an error", func() {
+				previousClusterConfig := &opv1alpha1.NetworkAddonsConfigSpec{
+					KubeMacPool: &opv1alpha1.KubeMacPool{RangeStart: "02:00:00:00:00:00", RangeEnd: "0A:FF:FF:FF:FF:FF"}}
+				currentClusterConfig := &opv1alpha1.NetworkAddonsConfigSpec{
+					KubeMacPool: &opv1alpha1.KubeMacPool{RangeStart: "02:00:00:ff:00:00", RangeEnd: "02:00:00:ff:00:02"}}
+				errorList := fillDefaultsKubeMacPool(currentClusterConfig, previousClusterConfig)
+				Expect(errorList).To(BeEmpty())
+				Expect(currentClusterConfig.KubeMacPool.RangeStart).To(Equal(currentClusterConfig.KubeMacPool.RangeStart))
```

Comment:
you send a pointer here so it will always be true.

please save the start and end range to environment variables and use it when you create the current config and also to compare here